chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -133,6 +133,9 @@ Shape GetInferredShape(const Shape& shape, int64_t total_size) {
         }
         int64_t rest_size = std::accumulate(inferred_shape.begin(), it, int64_t{1}, std::multiplies<>()) *
                             std::accumulate(std::next(it), inferred_shape.end(), int64_t{1}, std::multiplies<>());
+        if (rest_size == 0) {
+            throw DimensionError{"Cannot reshape array of size ", total_size, " into an ambiguous shape ", shape};
```

Comment:
The shape is ambiguous if the number of dimensions is larger than 1, the total size (product) is 0, but contains a negative value to be inferred.